clang/tools/cir-opt/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
+get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
+
+include_directories(${LLVM_MAIN_SRC_DIR}/../mlir/include)
+include_directories(${CMAKE_BINARY_DIR}/tools/mlir/include)
+
+add_clang_tool(cir-opt
+  cir-opt.cpp
+)
+
+clang_target_link_libraries(cir-opt
+  PRIVATE
+  clangCIR
+  clangCIRLoweringDirectToLLVM
+  MLIRCIR
```

Comment:
nit: `MLIRCIR` comes transitively from `clangCIR` no need for it here

Reply:
A shared library build fails with `_ZN4mlir6detail14TypeIDResolverIN3cir10CIRDialectEvE
2idE` as an unresolved symbol if I remove this.